src/app/app.component.ts

```diff
@@ -54,19 +54,29 @@ export class AppComponent {
     this.mode = event.target.value;
   }
 
-  public processImage(): void {
+  public async processImage(): Promise<void> {
     if (!this.imageFiles) {
       console.error('No image file selected.');
       return;
     }
 
-    this.imageFiles.forEach((file: File) => {
+    this.imageFiles.forEach(async (file: File) => {
       if (file) {
+        let blob: Blob = file;
+        // convert HEIC to JPEG if
+        if (file.type === 'image/heic') {
```

Comment:
Instead of requiring this as pre-check, can you incorporate this type of check and conversion into the actual load function?